torch/distributed/elastic/multiprocessing/__init__.py

```diff
@@ -220,25 +220,30 @@ def start_processes(
     error_files = {}
 
     for local_rank in range(nprocs):
-        clogdir = os.path.join(log_dir, str(local_rank))
-        os.mkdir(clogdir)
-
-        rd = redirs[local_rank]
-        if (rd & Std.OUT) == Std.OUT:
-            stdouts[local_rank] = os.path.join(clogdir, "stdout.log")
-        if (rd & Std.ERR) == Std.ERR:
-            stderrs[local_rank] = os.path.join(clogdir, "stderr.log")
-
-        t = ts[local_rank]
-        if t & Std.OUT == Std.OUT:
-            tee_stdouts[local_rank] = stdouts[local_rank]
-        if t & Std.ERR == Std.ERR:
-            tee_stderrs[local_rank] = stderrs[local_rank]
-
-        error_file = os.path.join(clogdir, "error.json")
-        error_files[local_rank] = error_file
-        log.info(f"Setting worker{local_rank} reply file to: {error_file}")
-        envs[local_rank]["TORCHELASTIC_ERROR_FILE"] = error_file
+        if log_dir == os.devnull:
+            tee_stdouts[local_rank] = os.devnull
+            tee_stderrs[local_rank] = os.devnull
+            error_files[local_rank] = os.devnull
```

Comment:
There are also other variables initialized in the "else" branch, `stdouts[local_rank]`, `stderrs[local_rank]`, `envs[local_rank]["TORCHELASTIC_ERROR_FILE"]`. Do we need to initialize here too?

Reply:
stdouts[local_rank] and stderrs[local_ranl] are initialized in https://github.com/pytorch/pytorch/blob/master/torch/distributed/elastic/multiprocessing/__init__.py#L216-L217. envs[local_rank]["TORCHELASTIC_ERROR_FILE"] is added in [4974e1e](https://github.com/pytorch/pytorch/pull/88477/commits/4974e1e85ec636dced8e70add33805e300dc6e19)